src/main/java/com/minecolonies/coremod/util/BarbarianUtils.java

```diff
@@ -0,0 +1,207 @@
+                    numberOfChiefBarbarians = hordeTotal - MAX_SIZE;
+
+                    //For error handling and correct hordeTotal
+                    if (numberOfChiefBarbarians < 0)
+                    {
+                        numberOfChiefBarbarians = 0;
+                    }
+                }
+            }
+        }
+
+        //Make sure world isn't null
+        if (raidingWorld == null)
+        {
+            return;
+        }
+
+        int x = colony.getCenter().getX();
+        int y = colony.getCenter().getY();
+        int z = colony.getCenter().getZ();
+
+        switch (raidingWorld.rand.nextInt(NUMBER_OF_POSSIBLE_CASES))
+        {
+            case 0:
+                x += Configurations.workingRangeTownHall + NUMBER_OF_BLOCKS_OUTSIDE_RADIUS;
+                break;
+            case 1:
+                x -= Configurations.workingRangeTownHall + NUMBER_OF_BLOCKS_OUTSIDE_RADIUS;
+                break;
+            case 2:
+                z += Configurations.workingRangeTownHall + NUMBER_OF_BLOCKS_OUTSIDE_RADIUS;
+                break;
+            case 3:
+                z -= Configurations.workingRangeTownHall + NUMBER_OF_BLOCKS_OUTSIDE_RADIUS;
+                break;
+            case 4:
+                x += Configurations.workingRangeTownHall + NUMBER_OF_BLOCKS_OUTSIDE_RADIUS;
+                z += Configurations.workingRangeTownHall + NUMBER_OF_BLOCKS_OUTSIDE_RADIUS;
+                break;
+            case 5:
+                x += Configurations.workingRangeTownHall + NUMBER_OF_BLOCKS_OUTSIDE_RADIUS;
+                z -= Configurations.workingRangeTownHall + NUMBER_OF_BLOCKS_OUTSIDE_RADIUS;
+                break;
+            case 6:
+                x -= Configurations.workingRangeTownHall + NUMBER_OF_BLOCKS_OUTSIDE_RADIUS;
+                z += Configurations.workingRangeTownHall + NUMBER_OF_BLOCKS_OUTSIDE_RADIUS;
+                break;
+            case 7:
+                x -= Configurations.workingRangeTownHall + NUMBER_OF_BLOCKS_OUTSIDE_RADIUS;
+                z -= Configurations.workingRangeTownHall + NUMBER_OF_BLOCKS_OUTSIDE_RADIUS;
+                break;
+            default:
+                x += Configurations.workingRangeTownHall + NUMBER_OF_BLOCKS_OUTSIDE_RADIUS;
+                break;
+        }
+
+        //Make sure mob spawns on surface.
+        y = raidingWorld.getTopSolidOrLiquidBlock(new BlockPos.MutableBlockPos(x, y, z)).getY();
+
+        spawn(BarbarianUtils.barbarian, numberOfBarbarians, x, y, z, raidingWorld);
+        spawn(BarbarianUtils.archer, numberOfArcherBarbarians, x, y, z, raidingWorld);
+        spawn(BarbarianUtils.chief, numberOfChiefBarbarians, x, y, z, raidingWorld);
+    }
+
+    /**
+     * Spawns EntityToSPawn at the X, Y, Z
+     */
+
+    public static void spawn(final ResourceLocation entityToSpawn, final int numberOfSpawns, final int x, final int y, final int z, final World world)
+    {
+        IntStream.range(0, numberOfSpawns).forEach(theInteger ->
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this lambda expression number of lines from 27 to at most 20. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)